src/wstool/config_yaml.py

```diff
@@ -355,7 +355,10 @@ def get_path_spec_from_yaml(yaml_dict):
             elif key == "uri":
                 uri = value
             elif key == "version":
-                version = value
+                if value is not None: 
+                    version = str(value)
```

Comment:
Additionally, you could add ```str(arg).strip``` in the place the exception happened in vcstools, which should not be vulnerable to this. Possibly check other places we use strip without being careful about the type.

Reply:
So I addressed the issue here because it seemed to be caused due to the underlying `PyYaml` versions having different behavior in the way it returned a specific yawl value. So I could fix all exceptions related to `strip` or I could enforce that version is a string, which already something something above the file parsing layer already expected, so I assume there are other parts that depend on the same expectation of type.

Reply:
Sure, here is a good place for a fix. vcstools is a library in it's own right (potentially to be used by other projects), so if you have time, you could also make it less vulnerable, but it's a different story from this PR.

Reply:
Cool, I will take a look then to see what I can do with vcstools when I get a chance.